CactbotOverlay/FFXIVProcess.cs

```diff
@@ -2,1261 +2,1559 @@
+         PLD = 19,
+         MNK = 20,
+         WAR = 21,
+         DRG = 22,
+         BRD = 23,
+         WHM = 24,
+         BLM = 25,
+         ACN = 26,
+         SMN = 27,
+         SCH = 28,
+         ROG = 29,
+         NIN = 30,
+         MCH = 31,
+         DRK = 32,
+         AST = 33,
+         SAM = 34,
+         RDM = 35,
+         BLU = 36,
+         GNB = 37,
+         DNC = 38,
+      };
+
+      static bool IsGatherer(EntityJob job) {
+         return job == EntityJob.FSH || job == EntityJob.MIN || job == EntityJob.BTN;
+      }
+
+      static bool IsCrafter(EntityJob job) {
+         return job == EntityJob.CRP ||
+           job == EntityJob.BSM ||
+           job == EntityJob.ARM ||
+           job == EntityJob.GSM ||
+           job == EntityJob.LTW ||
+           job == EntityJob.WVR ||
+           job == EntityJob.ALC ||
+           job == EntityJob.CUL;
+      }
+
+      [Serializable]
+      public class EntityData {
+
+         public string name;
+         public uint id = 0;
+         public EntityType type = EntityType.None;
+         public ushort distance = 0;
+         public float pos_x = 0;
+         public float pos_y = 0;
+         public float pos_z = 0;
+         public int hp = 0;
+         public int max_hp = 0;
+         public int mp = 0;
+         public int max_mp = 0;
+         public short gp = 0;
+         public short max_gp = 0;
+         public short cp = 0;
+         public short max_cp = 0;
+         public EntityJob job = EntityJob.None;
+         public short level = 0;
+         public string debug_job;
+         public int shield_value = 0;
+         public int bait = 0;
+
+         public override bool Equals(object obj) {
+            return obj is EntityData && (EntityData)obj == this;
+         }
+
+         public override int GetHashCode() {
+            int hash = 17;
+            hash = hash * 31 + name.GetHashCode();
+            hash = hash * 31 + id.GetHashCode();
+            hash = hash * 31 + type.GetHashCode();
+            hash = hash * 31 + distance.GetHashCode();
+            hash = hash * 31 + pos_x.GetHashCode();
+            hash = hash * 31 + pos_y.GetHashCode();
+            hash = hash * 31 + pos_z.GetHashCode();
+            hash = hash * 31 + hp.GetHashCode();
+            hash = hash * 31 + max_hp.GetHashCode();
+            hash = hash * 31 + mp.GetHashCode();
+            hash = hash * 31 + max_mp.GetHashCode();
+            hash = hash * 31 + gp.GetHashCode();
+            hash = hash * 31 + max_gp.GetHashCode();
+            hash = hash * 31 + cp.GetHashCode();
+            hash = hash * 31 + max_cp.GetHashCode();
+            hash = hash * 31 + job.GetHashCode();
+            hash = hash * 31 + level.GetHashCode();
+            hash = hash * 31 + shield_value.GetHashCode();
+            hash = hash * 31 + debug_job.GetHashCode();
+            hash = hash * 31 + bait.GetHashCode();
+            return hash;
+         }
+      };
+
+      // EntityStruct {
```

Comment:
I think these comments could be deleted now that the class describes them itself.